zerver/tests/test_realm_linkifiers.py

```diff
@@ -210,15 +203,15 @@ def test_valid_urls(self) -> None:
             "https://example.com/!path",
             "https://example.com/foo.bar",
             "https://example.com/foo[bar]",
-            "https://example.com/%(foo)s",
-            "https://example.com/%(foo)s%(bars)s",
-            "https://example.com/%(foo)s/and/%(bar)s",
-            "https://example.com/?foo=%(foo)s",
+            "https://example.com/{foo}",
+            "https://example.com/{foo}{bars}",
+            "https://example.com/{foo}/and/{bar}",
+            "https://example.com/?foo={foo}",
```

Comment:
We might as well throw in some of the nontrivial URI Template features here.

```suggestion
            "https://example.com/{foo}/and{/bar}",
            "https://example.com/{?foo}",
```